src/raymath.h

```diff
@@ -707,9 +717,23 @@ RMAPI float Vector3LengthSqr(const Vector3 v)
 // Calculate two vectors dot product
 RMAPI float Vector3DotProduct(Vector3 v1, Vector3 v2)
 {
-    float result = (v1.x*v2.x + v1.y*v2.y + v1.z*v2.z);
-
-    return result;
+#if defined(__SSE2__) && defined(RL_USE_SIMD)
+    __m128 vecA = _mm_set_ps(0.0f, v1.z, v1.y, v1.x);
+    __m128 vecB = _mm_set_ps(0.0f, v2.z, v2.y, v2.x);
+
+    __m128 mul = _mm_mul_ps(vecA, vecB);
+
+  #if defined(__SSE3__)
+    __m128 sum = _mm_hadd_ps(mul, mul);
+    sum = _mm_hadd_ps(sum, sum);
+  #else // Non __SSE3__
+    __m128 sum1 = _mm_add_ps(mul, _mm_shuffle_ps(mul, mul, _MM_SHUFFLE(2, 3, 0, 1)));
+    __m128 sum = _mm_add_ps(sum1, _mm_shuffle_ps(sum1, sum1, _MM_SHUFFLE(1, 0, 3, 2)));
+  #endif
+    return _mm_cvtss_f32(sum);
+#else // Non SIMD
+    return v1.x*v2.x + v1.y*v2.y + v1.z*v2.z;
+#endif
```

Comment:
Just for the sake of structural consistency with the other functions pattern ([1](https://github.com/raysan5/raylib/blob/93a1e75741175b9a2136593064b3714e79a363e2/src/raymath.h#L718-L725), [2](https://github.com/raysan5/raylib/blob/93a1e75741175b9a2136593064b3714e79a363e2/src/raymath.h#L731-L738), [3](https://github.com/raysan5/raylib/blob/93a1e75741175b9a2136593064b3714e79a363e2/src/raymath.h#L744-L751), etc), could you please use:
```
float result = 0.0f;

// your #if code...

return result;
```

Reply:
> Just for the sake of structural consistency with the other functions pattern ([1](https://github.com/raysan5/raylib/blob/93a1e75741175b9a2136593064b3714e79a363e2/src/raymath.h#L718-L725), [2](https://github.com/raysan5/raylib/blob/93a1e75741175b9a2136593064b3714e79a363e2/src/raymath.h#L731-L738), [3](https://github.com/raysan5/raylib/blob/93a1e75741175b9a2136593064b3714e79a363e2/src/raymath.h#L744-L751), etc), could you please use:
> 
> ```
> float result = 0.0f;
> 
> // your #if code...
> 
> return result;
> ```

No problem!

Reply:
Thank you.